### PR TITLE
Disable DFCO when clutch is applied

### PR DESCRIPTION
**Pull request by alrijleh**
Allow clutchUp switch to be a gating term for fuel cut activation.  Enables feature requested in https://github.com/FOME-Tech/fome-fw/issues/22

### CODE REVIEW

File: firmware/controllers/sensors/sensor_type.h [hunk 1]
```diff
@@ -71,6 +71,8 @@ enum class SensorType : unsigned char {
 
 	BarometricPressure,
 
+	ClutchUp,
```
Comment (mck1117) [inline firmware/controllers/sensors/sensor_type.h:74]: unrelated?

File: firmware/controllers/algo/defaults/default_fuel.cpp [hunk 2]
```diff
@@ -109,6 +109,7 @@ static void setDefaultFuelCutParameters() {
 	engineConfiguration->coastingFuelCutTps = 2;
 	engineConfiguration->coastingFuelCutMap = 30;
 	engineConfiguration->coastingFuelCutClt = 60;
+	engineConfiguration->disableFuelCutOnClutch = false;
```
Comment (mck1117) [inline firmware/controllers/algo/defaults/default_fuel.cpp:112]: By the way, you don't have to set any defaults that have a bit value of 0. The configuration is zeroed before reset, so everything defaults to 0/false/etc.

File: firmware/controllers/sensors/sensor_type.h (hunk 1 above)
Comment (alrijleh) [reply firmware/controllers/sensors/sensor_type.h:74]: Good catch.  Initially thought I needed to plumb it through before I saw that the digital inputs were getting set in engine.cpp.  This can/should be removed
